Marlin/ultralcd.cpp

```diff
@@ -830,14 +831,13 @@ static void _lcd_move(const char* name, AxisEnum axis, int min, int max) {
     if (min_software_endstops) NOLESS(current_position[axis], min);
     if (max_software_endstops) NOMORE(current_position[axis], max);
     encoderPosition = 0;
-    if (movesplanned() <= 3)
-      line_to_current(axis);
```

Comment:
I don't see the difference in your solution. The moves you can't do immediately are accumulated in current_position. We do not leave the move axis menu so the in formation about what axis to move is stored in 'currentMenu'. As long as we turn or klick the encoder, it does not make a difference if we enqueue the next move at the beginning of `lcd_update()`, or a tiny bit later, when we are back in the move menu.